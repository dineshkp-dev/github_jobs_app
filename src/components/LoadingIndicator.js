import React from 'react';
import '../styles/loadingindicator.css';

function LoadingIndicator(props) {
  const loadingText =
    (props && props.text) || 'Please wait for the content to be loaded';
  return (
    <div className='loading-indicator-container'>
      <h3 className='loading-title'>Loading...</h3>
      <div className='loading-text'>{loadingText}</div>
    </div>
  );
}

export default LoadingIndicator;
