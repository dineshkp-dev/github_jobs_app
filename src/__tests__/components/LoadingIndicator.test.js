import LoadingIndicator from '../../components/LoadingIndicator';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

describe('LoadingIndicator snapshot test', () => {
  test('renders', () => {
    const { container } = render(<LoadingIndicator />);
    expect(container).toMatchSnapshot();
  });
});
