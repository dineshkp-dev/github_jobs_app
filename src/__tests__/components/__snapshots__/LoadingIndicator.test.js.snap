// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingIndicator snapshot test renders 1`] = `
<div>
  <div
    class="loading-indicator-container"
  >
    <h3
      class="loading-title"
    >
      Loading...
    </h3>
    <div
      class="loading-text"
    >
      Please wait for the content to be loaded
    </div>
  </div>
</div>
`;
